common/beerocks/bcl/include/bcl/network/socket.h

```diff
@@ -141,4 +145,35 @@ class SocketSelect {
     std::string m_error;
 };
 
+class SocketPoll {
+
+public:
+    // Socket handlers type
+    typedef const std::function<bool(std::shared_ptr<Socket>)> socket_handler_t;
+
+public:
+    // Default poll timeout is 500ms
+    SocketPoll(uint timeout = 500);
+    ~SocketPoll();
+
+    bool add_socket(std::shared_ptr<Socket> socket, uint32_t additional_flags = 0);
+    bool del_socket(std::shared_ptr<Socket> socket);
+
+    // @retval -1 - Error
+    // @retval  0 - Timeout
+    // @retval  n - Processed sockets
+    int poll(socket_handler_t data_handler, socket_handler_t error_handler = {},
+             socket_handler_t disconnect_handler = {});
```

Comment:
> It uses select instead of epoll, however.
It can still be used as inspiration. It will probably not be cut-and-pastable anyway because we use a lot of different support functionality, e.g. logging, plus different coding style.

Anyway, the "difficult" part is designing a good interface. Once you have the function declarations, implementing them is trivial. Well, only the timer queue implementation needs a bit of thinking, but that can probably be taken over as-is.

Reply:
I'll work with @mariomaz to integrate the reactor pattern as part of this PR.